core/src/main/java/com/netflix/iceberg/BaseFileScanTask.java

```diff
@@ -66,6 +70,15 @@ public Expression residual() {
     return residuals.residualFor(file.partition());
   }
 
+  @Override
+  public Iterable<FileScanTask> split(long splitSize) {
+    if (file.format().isSplittable()) {
+      return () -> new SplitScanTaskIterator(splitSize, this);
+    } else {
+      return Lists.newArrayList(this);
```

Comment:
I think it would be better to use `ImmutableList.of(this)`. It is good to return immutable objects even though this probably won't be mutated. And that `ImmutableList` can use an implementation that is more efficient than `ArrayList` because it knows that there is only going to be one item.

Reply:
Makes sense